### PR TITLE
adding docs for griptape-cloud-knowledge-base-client

### PR DESCRIPTION
**Pull request by emjay07**
Original Doc: https://docs.google.com/document/d/1BWW9wI-2cZg4ZPOIxZ6lms3uo5fEhxFvopybN3vNDbM/edit

----
📚 Documentation preview 📚: https://griptape--824.org.readthedocs.build//824/

### CODE REVIEW

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+## Overview
+
+The `GriptapeCloudKnowledgeBaseClient` is an advanced Tool designed to streamline how organizations access and search their data. It leverages a RAG pipeline and vector stores hosted in [Griptape Cloud](https://cloud.griptape.ai) and allows users to perform intuitive searches across a centralized [Knowledge Base](https://cloud.griptape.ai/knowledge-bases) that consolidates various data sources such as Confluence, Google Docs, and web pages. This integration empowers users to connect and search through information pools, making it easier to find relevant data quickly.
+
+Griptape utilizes this technology to enhance organizational efficiency and knowledge sharing. By connecting diverse documentation and data sources into a single searchable repository, the `GriptapeCloudKnowledgeBaseClient` helps facilitate faster decision-making while improving productivity.
```
Comment (collindutter) [inline docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:5]: This feels a little marketing-ey for the docs. Can we be a bit more direct and say it connects to the RAG service on GT Cloud?

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+**Note:** This tool requires a [Knowledge Base](https://cloud.griptape.ai/knowledge-bases) hosted in Griptape Cloud and an [API Key](https://cloud.griptape.ai/keys) for access.
+
+```python
+import os
+from griptape.structures import Agent
+from griptape.tools import GriptapeCloudKnowledgeBaseClient
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (collindutter) [inline docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:15]: Remove, not included in Griptape deps.

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+knowledge_base_client = GriptapeCloudKnowledgeBaseClient(
+    description="Contains information about the company and its operations",
+    api_key=os.environ["GT_CLOUD_API_KEY"],
+    knowledge_base_id=os.environ["GT_CLOUD_KB_ID"],
+    off_prompt=False
```
Comment (collindutter) [inline docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:21]: Can remove

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import os
+from griptape.structures import Agent
+from griptape.tools import GriptapeCloudKnowledgeBaseClient
+from dotenv import load_dotenv
+
+load_dotenv()
+
+knowledge_base_client = GriptapeCloudKnowledgeBaseClient(
+    description="Contains information about the company and its operations",
+    api_key=os.environ["GT_CLOUD_API_KEY"],
```
Comment (collindutter) [inline docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:19]: Rename to `GRIPTAPE_CLOUD_API_KEY` to use existing API key in integ tests.

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    knowledge_base_id=os.environ["GT_CLOUD_KB_ID"],
```
Comment (collindutter) [inline docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:20]: Need to add to github integ tests workflow. Coordinate with Zach to create KB in the same account as the existing API key.

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md (hunk 1 above)
Comment (emjay07) [reply docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:5]: Sure, will simplify.

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md (hunk 4 above)
Comment (emjay07) [reply docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:19]: We may want to be consistent. the cloud structure template/logic use `GT_CLOUD_API_KEY` and not `GRIPTAPE_CLOUD_API_KEY`.

Comment (emjay07) [reply docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md]: Ah, they all have INTEG in front of them as well. updated.

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md (hunk 5 above)
Comment (emjay07) [reply docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md]: Done

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md (hunk 4 above)
Comment (collindutter) [reply docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md]: They do not have `INTEG` in the actual examples. Agreed that we should be consistent but we use `GRIPTAPE_CLOUD_API_KEY` in all the doc examples. I think this is an ok divergence from the template.

Comment (emjay07) [reply docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md]: weird... all of the environment variables in the action all start w/ `INTEG`? How are the others working?

Comment (collindutter) [reply docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md]: They get mapped in [here](https://github.com/griptape-ai/griptape/blob/dc9e58592d0a7f2a3ab17a7861e353110a6f7924/.github/workflows/docs-integration-tests.yml#L24)

File: docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+## Overview
+
+The `GriptapeCloudKnowledgeBaseClient` is an advanced Tool to retrieve data from a RAG pipeline and vector stores hosted in [Griptape Cloud](https://cloud.griptape.ai). It performs searches across a centralized [Knowledge Base](https://cloud.griptape.ai/knowledge-bases) that can consist of various data sources such as Confluence, Google Docs, and web pages.
```
Comment (collindutter) [inline docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:3]: Nit: the _tool_ isn't all that advanced -- it's just a light client for the advanced _service_

Comment (collindutter) [inline docs/griptape-tools/official-tools/griptape-cloud-knowledge-base-client.md:3]: > It performs searches across a centralized
Similar, maybe reword to emphasize that this is a feature of the service, not the tool.

**Review by collindutter — APPROVED** [review]
